app/controllers/api/infrastructures_controller.rb

```diff
@@ -32,10 +41,23 @@ def authorize_by_username
     @infrastructure = Infrastructure.
       find_by_cluster_name(params[:cluster_name])
 
-    if @infrastructure.blank? 
-      render json: { success: false, errors: ["Unauthorized: infrastructure #{params[:cluster_name]} is not exists"], code: 401 }, status: :unauthorized and return 
+    if @infrastructure.blank?
+      render json: {
+        success: false, 
+        errors: ["Unauthorized: infrastructure #{params[:cluster_name]} is not exists"],
+        code: 401
+      }, status: :unauthorized and return 
     end
 
+    if @infrastructure.status == Infrastructure.statuses[:inactive]
```

Comment:
why don't use `infrastructure.active?`
also please only return 404

Reply:
Forgot to replace this with that method. This is written before `infrastructure.active?` method is created